sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/retryprocessingerror/RetryProcessingErrorMLS.java

```diff
@@ -71,21 +85,26 @@ private void processRecordsByPartition(
           }
 
         } catch (RuntimeException e) {
-          LOGGER.error(
-              "Error while handling record {} in message handler {}",
-              consumerRecord.key(),
-              handler.getClass(),
-              e);
-          boolean shouldContinue = errorHandler.handleError(consumerRecord, e, consumer);
-          if (!shouldContinue) {
-            throw new StopListenerException(e);
+          long retryCount = getOffsetCounter(partition, consumerRecord.offset()).inc();
+          if (retryCount > maxRetryCount) {
+            LOGGER.error(
+                "Error while handling record {} in message handler {}, no more retries",
+                consumerRecord.key(),
+                handler.getClass(),
+                e);
+
+            callErrorHandler(finalErrorHandler, consumerRecord, e, consumer);
+            lastCommitOffset = markConsumerRecordProcessed(consumerRecord);
           } else {
             LOGGER.warn(
-                "Error while handling record {} in message handler {}, will be retried",
+                "Error while handling record {} in message handler {}, will be retried (%s / %s)..."
+                    .formatted(retryCount, maxRetryCount),
                 consumerRecord.key(),
                 handler.getClass(),
```

Comment:
Would it work with normal replacement of SLF4J as well?
```suggestion
                "Error while handling record {} in message handler {}, will be retried ({} / {})...",
                consumerRecord.key(),
                handler.getClass(),
                retryCount,
                maxRetryCount,
```

Reply:
Good point, done